File: hw/cañarte/u3/HW28SingletonAndAbstractFactory/GUIFactory/src/main/java/ec/edu/espe/guifactory/controller/LinuxFactory.java
package ec.edu.espe.guifactory.controller;

/**
 *
 * @author Saray Canarte, Code Creators, DCCO-ESPE
 */
public class LinuxFactory extends GUIFactory {
    @Override
    public Button createButton() {
        return new LinuxButton();
    }

    @Override
    public Menu createMenu() {
        return new LinuxMenu();
    }
}

File: hw/cañarte/u3/HW28SingletonAndAbstractFactory/Python/AbstractFactory/controller/LinuxFactory.py
import sys
import os
# Saray Cañarte, codecreators, DCCO - ESPE

sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
from controller.GUIFactory import GUIFactory
from controller.LinuxButton import LinuxButton
from controller.LinuxMenu import LinuxMenu


class LinuxFactory(GUIFactory):
    def create_button(self) -> LinuxButton:
        return LinuxButton()

    def create_menu(self) -> LinuxMenu:
        return LinuxMenu()
